File: tools/declare-csharp/testdata/events.cs
// Game-shaped C# event declarations for the labelle-declare-csharp byte-parity
// pin (tests/declare_csharp_tool.zig). No `using` line — the declare surface is
// global (a green run proves the tool needs no prelude injection). The
// declarations below are the golden's `csharp_events_source` verbatim
// (declare_cross_golden.zig).
[LabelleEvent]
record hunger__feed
{
    public ulong entity = 0;
    public double amount = 0.5;
    public bool urgent = false;
    public string reason = "why \"now\"";
    public Vec2 at = new(-1.5, 3.0);
}

[LabelleEvent]
record wave__spawned;

File: tools/declare-rs/testdata/events.rs
// A GAME-SHAPED event declaration file for the cross-runner byte-parity test
// (labelle-engine#774): bare `labelle::event!{…}` / `vec2(…)`, NO `use` lines —
// what a real game events/*.rs writes; the tool injects the prelude. The
// assembler passes components/*.rs before events/*.rs, so this file is argv
// index 1 (staged decl_0001.rs) and its events sort after the component file's
// — matching emit_schema's per-kind (file, line) order. Kept byte-in-sync with
// tests/declare_cross_golden.zig's `rust_events_source`.
labelle::event! {
    hunger__feed {
        entity: u64 = 0,
        amount: f32 = 0.5,
        urgent: bool = false,
        reason: str = "why \"now\"",
        at: vec2 = vec2(-1.5, 3.0),
    }
}

labelle::event! {
    wave__spawned {}
}
